### PR TITLE
feat: implement filtering of todos

### PR DESCRIPTION
**Pull request by beeman**

### CODE REVIEW

File: src/app/features/todos/todos.store.ts [hunk 1]
```diff
@@ -1,30 +1,50 @@
...
 import { Todo } from './models/todo'
 import { TodosService } from './todos.service'
+import { FormGroup } from '@angular/forms'
+import { UiFormField } from '../../ui/form/ui-form-field'
 
 interface TodosState {
   loading: boolean
   saving: boolean
   todos?: Todo[]
+  filter?: string | undefined
```
Comment (nartc) [inline src/app/features/todos/todos.store.ts:13]: suggestion: `filter?` implicitly has `string | undefined` already? Is this needed?

File: src/app/features/todos/todos.store.ts [hunk 2]
```diff
@@ -1,30 +1,50 @@
...
+      isLoading: loading && !todos?.length,
+      todos: filteredTodos,
+      saving,
+      filter,
+      form: this.form,
+      fields: this.fields,
+    }
+  })
+
+  updateFilter = this.effect<string>((query$) => query$.pipe())
```
Comment (nartc) [inline src/app/features/todos/todos.store.ts:47]: suggestion: I would make `updateFilter` into an `updater` instead.

```ts
this.updateFilter(this.query)

updateFilter = this.updater<string>((state, filter) => ({...state, filter}))
```

File: src/app/features/todos/todos.store.ts (hunk 1 above)
Comment (beeman) [reply src/app/features/todos/todos.store.ts:13]: Fixed

File: src/app/features/todos/todos.store.ts (hunk 2 above)
Comment (beeman) [reply src/app/features/todos/todos.store.ts]: Updated. Way better indeed 👍

**Pull request merged**